src/compiler/crystal/tools/init.cr

```diff
@@ -145,6 +137,20 @@ module Crystal
       args.shift
     end
 
+    def self.validate_name(name)
+      case
+      when name.blank?                       then raise Error.new("NAME must not be empty")
+      when name.size > 50                    then raise Error.new("NAME must not be longer than 50 characters")
+      when name.each_char.any?(&.uppercase?) then raise Error.new("NAME should be all lower cased")
+      when name.index("crystal")             then raise Error.new("NAME should not contain 'crystal'")
```

Comment:
This must go, there's no reason to error on this case.

Reply:
I agree: https://github.com/mverzilli/crystalla

(not active, but still...)

Reply:
Totally agree. I was just waiting on responses to my question in https://github.com/crystal-lang/crystal/pull/8737#issuecomment-583173458